tests/mdev_configuration_allocation_test.go

```diff
@@ -36,9 +36,24 @@ var _ = Describe("[Serial][sig-compute]MediatedDevices", func() {
 		virtClient, err = kubecli.GetKubevirtClient()
 		util.PanicOnError(err)
 	})
-	checkAllMDEVCreated := func(mdevTypeName string, expectedInstancesCount int) {
-		By(fmt.Sprintf("Checking the number of created mdev types, should be %d of %s type ", expectedInstancesCount, mdevTypeName))
-		check := fmt.Sprintf(`set -x
+
+	waitForPod := func(outputPod *k8sv1.Pod, fetchPod func() (*k8sv1.Pod, error)) wait.ConditionFunc {
+		return func() (bool, error) {
+
+			latestPod, err := fetchPod()
```

Comment:
this is a side-effect; it is mutating a variable defined on an outer scope.

Reply:
What do you mean?

Reply:
I think I'm wrong; I glanced at this and I thought it was mutating the `latestPod` variable defined in the outer scope in https://github.com/kubevirt/kubevirt/pull/8123/files#diff-a85c28d775a02e1d055f1eb1c7a0cf238646606a80e1403ed2ae198d5eab055aR74. I missed the `:=` assignment operator.

But, it does mutate the `outputPod` variable being passed to it. that's a side effect of the function - this function should "block until the pod is the expected phase". Or I expect that from its name - but it is updating one of the passed parameters, and doing what I expect it to do. That's surprising. I would suggest your update the variable outside explicitly, which makes the code easier to read and reason about.

Reply:
@maiqueb Hi. I can't see a trivial refactoring solution here, I would take the suggested changes since we need the functionality for tests stability. We can refactor this later on.